lamindb/history/_trigger_installer.py

```diff
@@ -0,0 +1,564 @@
+import enum
+import re
+from abc import ABC, abstractmethod
+from typing import Any
+
+from django.db import models, transaction
+from django.db.backends.base.base import BaseDatabaseWrapper
+from django.db.backends.utils import CursorWrapper
+from lamin_utils import logger
+from typing_extensions import override
+
+from lamindb.models.history import HistoryLock, HistoryMigrationState, HistoryTableState
+
+from ._db_metadata_wrapper import (
+    DatabaseMetadataWrapper,
+    PostgresDatabaseMetadataWrapper,
+)
+from ._types import KeyConstraint, UIDColumns
+
+
+class HistoryEventTypes(enum.Enum):
+    INSERT = 0
+    UPDATE = 1
+    DELETE = 2
+
+
+class DjangoMigration(models.Model):
```

Comment:
Note for @falexwolf, potentially de-duplicate with the below, but for now better to keep `history` self-contained.
 https://github.com/laminlabs/lamindb/blob/6537063258d341d11606a73cf964584f2c1092d4/lamindb/models/record.py#L1654